### ISSUE TITLE
JWE Decode with JweAlgorithm.RSA_OAEP_256 and JweEncryption.A256GCM using .net Core 2.0 on Linux

### ISSUE THREAD

**Issue by maximeMD** [dvsekhvalnov/jose-jwt#107]
Hello dear developer(s) 😄 

**First of all, this issue is the next step of this topic** : https://github.com/dvsekhvalnov/jose-jwt/issues/106
(sorry if re-opening the topic was a better practice, I am not for the moment very familiar with GitHub)

We managed with @matthias-jer2spy to make our snippet work on Windows :

```
public static string GetJws(string jwe, string decrypterPrivateFile, string decrypterPrivatePwd, string chPublic)  
{
   TextReader decrypterPrivateReader = System.IO.File.OpenText(decrypterPrivateFile);
   PemReader decrypterPrivatePemReader = new PemReader(decrypterPrivateReader, new    PasswordFinder(decrypterPrivatePwd));
   RsaPrivateCrtKeyParameters decrypterPrivateKey = (RsaPrivateCrtKeyParameters)((AsymmetricCipherKeyPair)decrypterPrivatePemReader.ReadObject()).Private;

   RSAParameters decrypterPrivateParams = DotNetUtilities.ToRSAParameters(decrypterPrivateKey);
   RSACng decrypterPrivateProvider = new RSACng();
   decrypterPrivateProvider.ImportParameters(decrypterPrivateParams);

   string jws = Jose.JWT.Decode(jwe, decrypterPrivateProvider, JweAlgorithm.RSA_OAEP_256, JweEncryption.A256GCM);
   return jws
}
```

After observing that the RSACng object is only working on Windows, we modified this object to:
`RSA decrypterPrivateProvider = RSA.Create(decrypterPrivateParams);`

Unfortunately, when we are pushing this code to our Linux server, we get (again) the following error : 
**Specified padding mode is not valid for this algorithm.**

We tried to change the line 
`RSA decrypterPrivateProvider = RSA.Create(decrypterPrivateParams);` 
to
`RSAOpenSsl decrypterPrivateProvider = new RSAOpenSsl(decrypterPrivateParams);` 
because the JOSE documentation specified that RSAOpenSsl was the one working on Linux.
It doesn't work either. Same error.

We also tried  .Decode() without explicit algs. Doesn't work either.

Do you have maybe any clues ? We already lost a tremendous amount of time (and anger) on this issue...

Thank you in advance for your help 😄 

PS : On the JOSE documentation, I think there is a little mistake on the classname : 
_Types returned by crytographic methods MAY be different on Windows and Linux. e.g. GetRSAPrivateKey() on X509Certificate2 on Windows returns RsaCng and **RSAOpenSsl** on *nix._

**Comment by dvsekhvalnov** [comment]
Hi @maximeMD ,

`X509Certificate2. GetRSAPrivateKey()` should be enough. If you on .NET 4.6+ or .netcore you don't need to work with concrete implementation classes (RSACng or RSAOpenSsl).

The thing is, AES-GCM will not work on linux at this time, given it's simply not implemented in .netcore.
See https://github.com/dvsekhvalnov/jose-jwt/issues/45 for details.

TL;DR: managed AES-GCM coming to .net as of .netstandard v3 release. Once it supported, i'll make changes to library. Hope it will make linux support better.

**Comment by maximeMD** [comment]
Hi @dvsekhvalnov ,

Thank you for your answer !

For the record, and if people are passing by here having the same issue, I built a little Node.js service only to decode the jws, and i am calling this from my .NET Core application. This works like a charm, event if it is a quite messy way to solve the problem ...
